github/pkg/adapter/adapter_test.go

```diff
@@ -402,236 +430,115 @@ var testCases = []testCase{
-	}
-
-	h := &fakeHandler{
-		//handler: tc.sink,
-		handler: sinkAccepted, // No tests expect the sink to do anything interesting
-	}
-	sinkServer := httptest.NewServer(h)
-	defer sinkServer.Close()
-
-	ra, err := New(sinkServer.URL, testOwnerRepo)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	payload := gh.PullRequestPayload{}
-	subject, _ := strconv.ParseInt(testSubject, 10, 64)
-	payload.PullRequest.Number = subject
-	header := http.Header{}
-	header.Set("X-"+GHHeaderEvent, eventType)
-	header.Set("X-"+GHHeaderDelivery, eventID)
-	ra.HandleEvent(payload, http.Header(header))
-
-	// TODO(https://knative.dev/pkg/issues/250): clean this up when there is a shared test client.
-
-	canonicalizeHeaders(expectedRequest)
-	if diff := cmp.Diff(expectedRequest.Headers, h.header); diff != "" {
-		t.Errorf("Unexpected difference (-want, +got): %v", diff)
-	}
 
-	if diff := cmp.Diff(expectedRequest.Body, string(h.body)); diff != "" {
-		t.Errorf("Unexpected difference (-want, +got): %v", diff)
+	eventID := ce.Sent()[0].ID()
+	if eventID != tc.eventID {
+		t.Fatalf("Expected %q event id to be sent, got %q", tc.eventID, eventID)
 	}
-}
-
-type fakeHandler struct {
-	body   []byte
-	header http.Header
+	got := ce.Sent()[0].Data()
 
-	handler func(http.ResponseWriter, *http.Request)
-}
-
-func (h *fakeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	body, err := ioutil.ReadAll(r.Body)
+	data, err := json.Marshal(tc.payload)
```

Comment:
If it's a json always avoid byte array/string comparisons since the serialization order of the fields could be not deterministic, so the assertion below `string(got) != string(data)` could fail even if the JSONs are semantically equivalent.

Parse both tc.payload and sent event data into `interface{}` and then use `reflect.DeepEqual` or `cmp.Diff`:

```
if diff := cmp.Diff(want, got); diff != "" {
 t.Errorf("unexpected data (-want, +got) = %v", diff)
}
```

Reply:
Done.

Reply:
You didn't parsed the json:

```
var want interface{}
json.Unmarshal(tc.payload, &want)
```

etc...